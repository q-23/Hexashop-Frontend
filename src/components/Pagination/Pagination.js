import React from "react";

import BoxHeaderContainer from "components/_Shared/BoxHeaderContainer";
import {PaginationArrowButton, PaginationButton} from "components/Pagination/Pagination.style";
import FlexContainer from "components/_Shared/FlexContainer";
import Icon from "components/_Shared/Icon";

const Pagination = ({ currentPage = 1, nrOfPages = 1 }) => {

	const paginate = (currentPage, nrOfPages) => {
		const delta = 2;
		const	range = [];
		const rangeWithDots = [];
		let l;

		range.push(1);

		if (nrOfPages <= 1) {
			return range;
		}

		for (let i = currentPage - delta; i <= currentPage + delta; i++) {
			if (i < nrOfPages && i > 1) {
				range.push(i);
			}
		}
		range.push(nrOfPages);

		for (let i of range) {
			if (l) {
				if (i - l === 2) {
					rangeWithDots.push(l + 1);
				} else if (i - l !== 1) {
					rangeWithDots.push('...');
				}
			}
			rangeWithDots.push(i);
			l = i;
		}
		return rangeWithDots;
	};

	return(
		<BoxHeaderContainer variant_down>
			<FlexContainer justify={'center'} align={'center'} styles={'height: 100%'}>
				<PaginationArrowButton>
					<Icon className={'fa fa-arrow-left'}/>
				</PaginationArrowButton>
				{
					paginate(currentPage, nrOfPages).map(el => {
						return (
							<PaginationButton dots={typeof el === 'string'}>
								{el}
							</PaginationButton>)
					})
				}
				<PaginationArrowButton>
					<Icon className={'fa fa-arrow-right'}/>
				</PaginationArrowButton>
			</FlexContainer>
		</BoxHeaderContainer>
	)
};

export default Pagination;