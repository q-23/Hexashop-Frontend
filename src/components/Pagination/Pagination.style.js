import styled from "styled-components";
import * as palette from 'assets/css_variables/colors';

const PaginationArrowButton = styled.button`
	border-radius: 50%;
	border: none;
	background-color: transparent;
	cursor: pointer;
	transition: .2s ease-in-out;
	font-size: 2rem;
	&:hover i {
		text-shadow: 0px 4px 10px ${palette.LIGHT_PURPLE};
		color: ${palette.DARK_GRAY};
	}
`;

const PaginationButton = styled.button`
	border: none;
	background-color: transparent;
	transition: .2s ease-in-out;
	text-decoration: ${({current}) => current ? 'underline' : 'none'};
	cursor: ${({ dots }) => dots ? 'unset' : 'pointer'};
	font-size: 1.2rem;
`;


export { PaginationArrowButton, PaginationButton };