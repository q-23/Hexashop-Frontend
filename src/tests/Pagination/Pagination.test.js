import Pagination from "components/Pagination";
import React from "react";
import { render } from '@testing-library/react'
import { RenderWithPagination } from "tests/utils";

describe('[PAGINATION]', () => {
	test('Should render one page button when given only one page', async () => {
		const wrapper = render(RenderWithPagination(<Pagination/>))

		expect(wrapper).toMatchSnapshot();
	});
})