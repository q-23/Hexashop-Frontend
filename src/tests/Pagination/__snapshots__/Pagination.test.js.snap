// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGINATION] Should render one page button when given only one page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm bnhQgg"
      >
        <div
          class="sc-AxhCb cmWvdN"
        >
          <button
            class="sc-AxirZ bcQSl"
          >
            <i
              class="sc-AxhUy egJAyM fa fa-arrow-left"
            />
          </button>
          <button
            class="sc-AxiKw iUMAXY"
          >
            1
          </button>
          <button
            class="sc-AxirZ bcQSl"
          >
            <i
              class="sc-AxhUy egJAyM fa fa-arrow-right"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm bnhQgg"
    >
      <div
        class="sc-AxhCb cmWvdN"
      >
        <button
          class="sc-AxirZ bcQSl"
        >
          <i
            class="sc-AxhUy egJAyM fa fa-arrow-left"
          />
        </button>
        <button
          class="sc-AxiKw iUMAXY"
        >
          1
        </button>
        <button
          class="sc-AxirZ bcQSl"
        >
          <i
            class="sc-AxhUy egJAyM fa fa-arrow-right"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
